tsfile/src/main/java/org/apache/iotdb/tsfile/write/record/Tablet.java

```diff
@@ -95,12 +102,56 @@ public Tablet(String deviceId, List<MeasurementSchema> schemas, int maxRowNumber
     this.deviceId = deviceId;
     this.schemas = schemas;
     this.maxRowNumber = maxRowNumber;
+    mschemawithindex = new HashMap<>();
+
+    for (int i = 0; i < schemas.size(); i++) {
+      mschemawithindex.put(schemas.get(i).getMeasurementId(), i);
+    }
 
     createColumns();
 
     reset();
   }
 
+  public void addTimestamp(int rowIndex, long timestamp) {
+    timestamps[rowIndex] = timestamp;
+  }
+
+  public void addValue(String measurementId, int rowIndex, Object value) {
+    int indexOfValue = mschemawithindex.get(measurementId);
+    MeasurementSchema measurementSchema = schemas.get(indexOfValue);
+
+    switch (measurementSchema.getType()) {
+      case TEXT:
+        Binary[] sensort = (Binary[]) values[indexOfValue];
```

Comment:
It's too ugly to create so many new variable names. 
`you could use {} in every case`

Reply:
ok, i will improve that